### DISCUSSION TITLE
Is there a way to reduce spam?

### DISCUSSION THREAD

**Discussion by Jeffrey04** [huan/docker-simple-mail-forwarder#70] [General]
when I go through the logs I see there are quite a number of spams going through the service, is there a way to somewhat block most, or at least some of it?

**Comment by JeffHiggy** [comment]
Just my 2 cents but the main reason I use this project is because it indiscriminately forwards everything, I can do the spam filtering on the receiving end where I can filter out to a spam folder vs just throwing out the email that may be a false positive.

There are similar projects that do quite a bit of spam checks.

**Comment by Jeffrey04** [comment]
my webhost block mail server entirely (hence my pull request to relay all mail sending work to my gmail smtp), so it would be nice to have some simple spam filtering done on server side so my gmail sent mail record get cleaner relatively

**Comment by xriser** [comment]
Hi, guys. I don't like fail2ban, so I have dug a little and found a nice feature out of the box - it is postscreen. http://www.postfix.org/POSTSCREEN_README.html
You can check the docs about, and my commits on my forked repo I did on Mar 7, 2021
https://github.com/xriser/docker-simple-mail-forwarder/commit/fc9a0f88c9ec57e3974bf6785ab46dfec3373ea9
In general, you need to change master.cf and main.cf to enable it.
There are few nice features like quick tests, deep tests with soft graylist, blacklisting, and whitelisting.
But be careful with deep tests with soft graylist, because for example gmail does retries from different IPs, so it needs to add trusted MTA to the whitelist. I am using postwhite for that. https://github.com/stevejenkins/postwhite
